ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,699 @@
+from abc import ABC
+from dataclasses import field
+from typing import Any, ClassVar, Optional, Union
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.constants import (
+    AUDIO,
+    BAG,
+    BINARY,
+    CATEGORY,
+    DATE,
+    H3,
+    IMAGE,
+    MISSING_VALUE_STRATEGY_OPTIONS,
+    NUMBER,
+    SEQUENCE,
+    SET,
+    TEXT,
+    TIMESERIES,
+    VECTOR,
+)
+from ludwig.schema import utils as schema_utils
+from ludwig.utils import strings_utils
+from ludwig.utils.registry import Registry
+from ludwig.utils.tokenizers import tokenizer_registry
+
+preprocessing_registry = Registry()
+
+
+def register_preprocessor(name: str):
+    def wrap(preprocessing_config: BasePreprocessingConfig):
+        preprocessing_registry[name] = preprocessing_config
+        return preprocessing_config
+
+    return wrap
+
+
+@dataclass
+class BasePreprocessingConfig(schema_utils.BaseMarshmallowConfig, ABC):
+    """Base class for input feature preprocessing. Not meant to be used directly.
+
+    The dataclass format prevents arbitrary properties from being set. Consequently, in child classes, all properties
+    from the corresponding input feature class are copied over: check each class to check which attributes are different
+    from the preprocessing of each feature.
+    """
+
+    feature_type: ClassVar[Optional[str]] = None
+    "Class variable pointing to the corresponding preprocessor."
+
+    type: str
+
+
+@dataclass
```

Comment:
If this isn't causing bugs then its a moot point, but it looks like you are registering the preprocessing configs before they are marshmallow-fied by the dataclass decorator.